ndcube/ndcube.py

```diff
@@ -381,24 +391,46 @@ def axis_world_coords(self, *axes):
                 if n_dependent_axes[i] == 1:
                     # Construct pixel quantities in each dimension letting
                     # other dimensions all have 0 pixel value.
-                    quantity_list = [
-                        u.Quantity(np.zeros(cube_dimensions[dependent_axes[i]]),
-                                   unit=u.pix)] * n_dimensions
                     # Replace array in quantity list corresponding to current axis with
                     # np.arange array.
-                    quantity_list[axis] = u.Quantity(np.arange(cube_dimensions[axis]), unit=u.pix)
+
+                    # Check for the edges arguments
+                    if edges:
+                        # Setting start and stop ranges for `pixel_edges`
+                        start = -0.5
+                        stop = cube_dimensions[axis] + 0.5
+                        quantity_list = [
+                        u.Quantity(np.zeros(cube_dimensions[dependent_axes[i]]+1),
+                                   unit=u.pix)] * n_dimensions
+                        quantity_list[axis] = u.Quantity(np.arange(start,stop), unit=u.pix)
+                    else:
+                        quantity_list = [
+                        u.Quantity(np.zeros(cube_dimensions[dependent_axes[i]]),
+                                   unit=u.pix)] * n_dimensions
+                        quantity_list[axis] = u.Quantity(np.arange(cube_dimensions[axis]), unit=u.pix)
                 else:
                     # If the axis is dependent on another, perform
                     # translations on all dependent axes.
                     # Construct pixel quantities in each dimension letting
                     # other dimensions all have 0 pixel value.
-                    quantity_list = [u.Quantity(np.zeros(tuple(
+                    # Construct orthogonal pixel index arrays for dependent axes.
+                    # Check for edges arguments
+                    if edges:
+                        quantity_list = [u.Quantity(np.zeros(tuple(
+                        [cube_dimensions[k]+1 for k in dependent_axes[i]])),
+                        unit=u.pix)] * n_dimensions
+
+                        dependent_pixel_quantities = np.meshgrid(
+                            *[np.arange(-0.5, cube_dimensions[k] + 0.5) * u.pix
+                              for k in dependent_axes[i]], indexing="ij")
+                    else:
+                        quantity_list = [u.Quantity(np.zeros(tuple(
                         [cube_dimensions[k] for k in dependent_axes[i]])),
                         unit=u.pix)] * n_dimensions
-                    # Construct orthogonal pixel index arrays for dependent axes.
```

Comment:
I think it's important to keep this comment.

Reply:
Okay.